From cc40aa4211b95c52052c3207155889d460146114 Mon Sep 17 00:00:00 2001
From: Chao Liu <chao.liu.zevorn@gmail.com>
Date: Sun, 4 Oct 2026 16:03:52 +0800
Subject: [PATCH] util: omit wall-clock timestamps on Zephyr

Keep error output available without GLib calendar objects. Omit
optional wall-clock timestamp formatting on the Zephyr host.

Signed-off-by: Chao Liu <chao.liu.zevorn@gmail.com>

---
 util/error-report.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/util/error-report.c b/util/error-report.c
index 8932bee..56cf5c3 100644
--- a/util/error-report.c
+++ b/util/error-report.c
@@ -210,11 +210,13 @@ static void print_loc(MonitorHMP *hmp)
     }
 }
 
+#ifndef CONFIG_ZEPHYR
 char *real_time_iso8601(void)
 {
     g_autoptr(GDateTime) dt = g_date_time_new_now_utc();
     return g_date_time_format_iso8601(dt);
 }
+#endif
 
 /*
  * Print a message to current HMP monitor if we have one, else to
@@ -228,17 +230,18 @@ G_GNUC_PRINTF(2, 0)
 static void vreport(report_type type, const char *fmt, va_list ap)
 {
     MonitorHMP *hmp = monitor_cur_hmp();
-    gchar *timestr;
 
     if (!hmp) {
         qemu_flockfile(stderr);
     }
 
+#ifndef CONFIG_ZEPHYR
     if (message_with_timestamp && !hmp) {
-        timestr = real_time_iso8601();
+        gchar *timestr = real_time_iso8601();
         fprintf(stderr, "%s ", timestr);
         g_free(timestr);
     }
+#endif
 
     /* Only prepend guest name if -msg guest-name and -name guest=... are set */
     if (error_with_guestname && error_guest_name && !hmp) {
